### PR TITLE
Add to cart

### PR DESCRIPTION
**Pull request by slava000**
Oops I think this might be a big PR, I'm sorry. 
@tpminhtam @xinxin510 @zacharyatha 

@tpminhtam I think it's your turn to review

**Styles**
The current selection should be indicated within the list by the overlay of a checkmark on top of the thumbnail for that style. Additionally, the title for that style should appear typed out in full above the thumbnail list.

A user will be able to change the selected style by clicking on the thumbnail displaying that style. Clicking on the thumbnail for the currently selected style will have no impact.

By default, the style selected will be the first in the list.

Only one style can be selected at a time. A style must be selected at all times.

**Size dropdown**
The first dropdown will list all of the available sizes for the currently selected style.  

Only sizes that are currently in stock for the style selected should be listed.  Sizes not available should not appear within the list.  

When collapsed, the dropdown should show the currently selected size.  

By default, the dropdown should show “Select Size”. 

**Quantity dropdown**
If the size has not been selected, then the quantity dropdown will display ‘-’ and the dropdown will be disabled.  

Once a size has been selected, the quantity dropdown should default to 1. 

The second dropdown will allow the user to select a quantity of the current style and size to add to their cart.  

The options in this dropdown will be a sequence of integers ranging from 1 to the maximum.  The maximum selection will be capped by either the quantity of this style and size in stock, or a hard limit of 15.   For example, if the SKU for the selected product style and size has 4 units left in stock, the dropdown will allow choice of 1, 2, 3 or 4.  However if there are 30 units in stock, the dropdown will only present from 1 to 15.

### CODE REVIEW

File: client/src/components/App.jsx [hunk 1]
```diff
@@ -20,7 +20,7 @@ export default class App extends React.Component {
   componentDidMount() {
     axios(`/overview/${this.state.product_id}`)
     .then(({data})=>{
-      // console.log(data);
+      console.log(data);
```
Comment (tpminhtam) [inline client/src/components/App.jsx:23]: Should we comment out the console.log?

File: server/server.js [hunk 2]
```diff
@@ -20,14 +20,32 @@ app.get('/overview/:product_id', (req, res) => {
...
+    headers: {Authorization: token}
+  };
+
+  axios.post(url, options)
+  .then((result)=>{
+    console.log('post result', result);
+
+  })
+  .catch((err)=>{
+    console.log('post cart err: ', err);
```
Comment (tpminhtam) [inline server/server.js:45]: Does your request return anything to the browser? It doesn't look like there is a `res.send`

**Review by tpminhtam — APPROVED** [review]

File: client/src/components/App.jsx (hunk 1 above)
Comment (slava000) [reply client/src/components/App.jsx:23]: yes

File: server/server.js (hunk 2 above)
Comment (slava000) [reply server/server.js:45]: i haven't gotten this route to work yet, will comment it out for now

**Pull request merged**